src/app/pages/press/article/article.js

```diff
@@ -1,52 +1,65 @@
-import bodyUnitView from '~/components/body-units/body-units';
-import Byline from '~/components/byline/byline';
+import React from 'react';
+import BodyUnit from '~/components/body-units/body-units.jsx';
+import {BylineJsx} from '~/components/byline/byline';
 import {formatDateForBlog as formatDate} from '~/helpers/data';
-import {description as template} from './article.html';
-import css from '~/pages/blog/article/article.css';
+import '~/pages/blog/article/article.css';
+import {usePageData} from '~/helpers/controller/cms-mixin';
 
-const spec = {
-    template,
-    css,
-    view: {
-        classes: ['article']
-    },
-    regions: {
-        body: '.body'
-    },
-    slug: 'setInInit',
-    model: {}
-};
-
-export default class Article extends componentType(spec) {
+function Hero({coverUrl}) {
+    return (
+        <div className="hero" style={`background-image: url(${coverUrl})`}>
+            <img className="strips" src="/images/components/strips.svg"
+                height="10" alt="" role="presentation" />
+        </div>
+    );
+}
 
-    init(slug) {
-        super.init();
-        this.slug = slug;
-        this.preserveWrapping = true;
-    }
+function Article({data}) {
+    const {
+        article_image: coverUrl,
+        subheading,
+        title,
+        author,
+        date: rawDate,
+        body: bodyData
+    } = data;
+    const date = formatDate(rawDate);
 
-    attachByline(author, date) {
-        const bylineRegion = this.regionFrom('.byline');
+    return (
+        <div className="article text-content">
+            {
+                Boolean(coverUrl) && <Hero coverUrl={coverUrl} />
+            }
+            <article>
+                <h1>{title}</h1>
+                {
+                    Boolean(subheading) && <h2>{subheading}</h2>
+                }
+                <BylineJsx author={author} date={date} />
+                <div className="body">
+                    {
+                        bodyData.map((unit) => <BodyUnit unit={unit} />)
+                    }
+                </div>
+            </article>
+        </div>
+    );
+}
 
-        bylineRegion.attach(new Byline({
-            date,
-            author
-        }));
-    }
+export function ArticleLoader({slug}) {
+    const [data, statusPage] = usePageData({slug, preserveWrapping: true});
 
-    onDataLoaded() {
-        this.model = {
-            coverUrl: this.pageData.article_image,
-            heading: this.pageData.heading,
-            subheading: this.pageData.subheading,
-            title: this.pageData.title
-        };
-        this.update();
-        this.attachByline(this.pageData.author, formatDate(this.pageData.date));
-        this.pageData.body.forEach((bodyUnit) => {
-            this.regions.body.append(bodyUnitView(bodyUnit));
-        });
+    if (data && data.error) {
+        routerBus.emit('navigate', '/404', {path: '/blog'}, true);
+        return null;
```

Comment:
do you know that it's a 404 anytime there's an error?  Seemed strange to me that it's not checking anything about the error, but maybe that's the only time it occurs

Reply:
That's the only error I have available to me. Effectively, if I can't get the data, it's not found.